server.go

```diff
@@ -213,8 +270,29 @@ func (s *Server) handleService(mainconn net.Conn, serverName, peeked string, isT
 	log.Println("handling connection from ", mainconn.RemoteAddr())
 
 	conn := GetConn(mainconn, peeked)
-	if err := s.forward(conn, service.Addr, remotePort); err != nil {
-		log.Printf("failed to forward traffic: %v\n", err)
+	if service.ClientSecret != "" {
+		// forward to an active connection
+		if err := s.forwardToActiveConnection(conn, s.activeConnections[service.ClientSecret]); err != nil {
+			return err
+		}
+	} else {
+		if err := s.forward(conn, service.Addr, remotePort); err != nil {
+			log.Printf("failed to forward traffic: %v\n", err)
+			return err
+		}
+	}
+	return nil
+
+}
+
+func (s *Server) forwardToActiveConnection(conn, activeConn net.Conn) error {
```

Comment:
`forwardToActiveConnection` and `forward` are nearly the same function, can't we just move the dial out of `forward` and delete `forwardToActiveConnection` ?